aloe/registry.py

```diff
@@ -214,8 +214,12 @@ def load(self, sentence, func):
         self.steps[step_re.pattern] = (step_re, func)
 
         try:
+            if not getattr(func, 'patterns', None):
+                func.patterns = set()
+
+            func.patterns.add(step_re.pattern)
             func.sentence = sentence
-            func.unregister = partial(self.unload, step_re.pattern)
+            func.unregister = partial(self.unload, *func.patterns)
```

Comment:
with the `extended_step` it can register the function to two different sentences:

```
@extended_step('I log in as user {STRING}')
def login(self, username):
```

Will register the `login` function with the following sentences (one for single quotes the other for double quotes):
```
I log in as user '([^']*)'
I log in as user "([^"]*)"
```